src/librustc_lint/builtin.rs

```diff
@@ -349,7 +349,7 @@ impl MissingDoc {
         id: Option<hir::HirId>,
```

Comment:
Does anything pass `None` for this? If not you could remove the `Option` and compute `article_and_desc` inside here from the `HirId`.

Reply:
Unfortunately, it is passed with a `Crate` down below, so for now I will keep the option.